src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -1566,43 +1326,38 @@ Object                 state
         //ThreadPool has per-appdomain managed queue of work-items. The VM is
         //responsible for just scheduling threads into appdomains. After that
         //work-items are dispatched from the managed queue.
-        private static bool QueueUserWorkItemHelper(WaitCallback callBack, Object state, ref StackCrawlMark stackMark, bool compressStack )
+        private static bool QueueUserWorkItemHelper(WaitCallback callBack, Object state, ref StackCrawlMark stackMark, bool compressStack)
         {
-            bool success =  true;
-
-            if (callBack != null)
+            if (callBack == null)
             {
-                        //The thread pool maintains a per-appdomain managed work queue.
-                //New thread pool entries are added in the managed queue.
-                //The VM is responsible for the actual growing/shrinking of 
-                //threads. 
-
-                EnsureVMInitialized();
+                throw new ArgumentNullException(nameof(WaitCallback));
```

Comment:
Certainly looks like it.  This is what was there before, but I can change it.  Need to check to see whether any existing CoreFx tests are verifying the name,